runtime/gpufortrt/src/gpufortrt_api_core.f90

```diff
@@ -292,4 +292,31 @@ function gpufortrt_use_device_c_impl(hostptr,condition,if_present) &
     resultptr = gpufortrt_use_device_c_impl(hostptr,opt_if_arg,opt_if_present_arg)
   end function
 
+  function gpufortrt_is_present(data_arg, bytes) result(isOnDeviceMemory) 
+    use iso_c_binding
+    !
+    type(*), target, dimension(..)::data_arg
+    integer(c_int),value,intent(in) :: bytes
+    !
+    logical::isOnDeviceMemory
+    !
+    interface
+      function gpufortrt_is_present_c_impl(data_arg, bytes) &
+        bind(c,name="gpufortrt_present") result(isOnDeviceMemory) 
+          use iso_c_binding
+          !
+          type(c_ptr), value::data_arg
+          integer(c_size_t),value :: bytes
+          !
+          type(c_ptr)::isOnDeviceMemory
+          !
+      end function
+    end interface
+    if(c_associated(gpufortrt_is_present_c_impl(c_loc(data_arg),int(bytes,kind=c_size_t)))) then
```

Comment:
I think you can make an assignment out of this. 
Please further do not use the camelCase for the result arg,
breaks with the OpenACC look and feel. 
You could also just write
```Fortran
logical :: gpufortrt_is_present

...

gpufort_is_present = c_associated(...)
```

and get rid of the `result(...)` alltogether.

Reply:
fixed in commit https://github.com/ROCmSoftwarePlatform/gpufort/pull/45/commits/59d4293c415a47f93867135dddb827fb1fd7a4ae